yogstation/code/modules/mob/living/carbon/human/human_movement.dm

```diff
@@ -5,10 +5,16 @@
 		update_movespeed()
 
 /mob/living/carbon/human/update_move_intent_slowdown()
-	var/turf/T = get_turf(src)
-	if(!T || !is_mining_level(T.z)) // If we can't get the turf, assume it's not on mining.
+	var/turf/player_turf = get_turf(src)
+
+	if(!player_turf || ! is_mining_level(player_turf.z)) // If we can't get the turf, assume it's not on mining.
 		return ..()
 
+	if(is_mining_level(player_turf.z)) // for icemoon 
```

Comment:
You shouldn't need this second check, since it already returns if it isn't on a mining level